BWT/BWT/BWT.cs

```diff
@@ -0,0 +1,154 @@
+﻿// <copyright file = "BWT.cs" author = "psiblvdegod" date = "2025">
+// under MIT license
+// </copyright>
+
+namespace BWT;
+
+using System.Collections;
+
+/// <summary>
+/// Contains methods, which allow do Burrows-Wheeler transform.
+/// </summary>
+public static class BWT
+{
+    /// <summary>
+    /// Tranforms string using Burrows-Wheeler algorithm.
+    /// </summary>
+    /// <param name="input">String which will be transformed.</param>
+    /// <returns>Transformed string and it's position in the table of shifts.</returns>
+    public static (string Output, int Position) Transform(string input)
+    {
+        ArgumentNullException.ThrowIfNullOrEmpty(input);
+
+        var shifts = GetShifts(input);
+        var output = new char[input.Length];
+        var position = 0;
+
+        for (var i = 0; i < shifts.Length; ++i)
+        {
+            if (shifts[i] == 0)
+            {
+                output[i] = input[^1];
+                position = i;
+                continue;
+            }
+
+            output[i] = input[shifts[i] - 1];
+        }
+
+        return (string.Concat(output), position);
+    }
+
+    /// <summary>
+    /// Detransforms string which has been transformed with Burrows-Wheeler algorithm.
+    /// </summary>
+    /// <param name="input">String which will be detransformed.</param>
+    /// <param name="position">Position of input string in the shifts table.</param>
+    /// <returns>Initial string.</returns>
+    public static string Detransform(string input, int position)
+    {
+        ArgumentNullException.ThrowIfNullOrEmpty(input);
+
+        ArgumentOutOfRangeException.ThrowIfGreaterThanOrEqual(position, input.Length);
+
+        ArgumentOutOfRangeException.ThrowIfLessThan(position, 0);
+
+        int alphabetPower = (int)Math.Pow(2, sizeof(char) * 8);
```

Comment:
nit: Мне кажется, так будет понятнее, что тут происходит:
```suggestion
        int alphabetPower = char.MaxValue - char.MinValue;
```
или, учитывая, что `char` беззнаковый,
```suggestion
        int alphabetPower = char.MaxValue;
```